src/panels/lovelace/hui-view.js

```diff
@@ -93,6 +120,23 @@ class HUIView extends PolymerElement {
     this._badges = [];
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+    if (!registeredDialog) {
+      registeredDialog = true;
+      registerEditCardDialog(this);
```

Comment:
I just had a genius idea.

Put this check and fire event INSIDE `showEditCardDialog`. It's great as we keep all logic inside 1 file.